iam-migrate.go

```diff
@@ -0,0 +1,76 @@
+//
+// MinIO Object Storage (c) 2022 MinIO, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package madmin
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+// ExportIAM makes an admin call to export IAM data
+func (adm *AdminClient) ExportIAM(ctx context.Context) (io.ReadCloser, error) {
+	path := fmt.Sprintf(adminAPIPrefix + "/export-iam")
+
+	resp, err := adm.executeMethod(ctx,
+		http.MethodGet, requestData{
+			relPath: path,
+		},
+	)
+	if err != nil {
+		closeResponse(resp)
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, httpRespToErrorResponse(resp)
+	}
+
+	if resp.Body == nil {
+		return nil, errors.New("body is nil")
+	}
+
+	return resp.Body, nil
+}
+
+// ImportIAM makes an admin call to setup IAM  from imported content
+func (adm *AdminClient) ImportIAM(ctx context.Context, content []byte) error {
+	path := fmt.Sprintf(adminAPIPrefix + "/import-iam")
+
+	resp, err := adm.executeMethod(ctx,
+		http.MethodPut, requestData{
+			relPath: path,
+			content: content,
+		},
+	)
+	if err != nil {
+		closeResponse(resp)
+		return err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return httpRespToErrorResponse(resp)
+	}
+
+	if resp.Body == nil {
+		return errors.New("body is nil")
+	}
+
```

Comment:
use `defer closeResponse(resp)`  

closeResponse is not needed when executeMethod() actually fails since that failure is when resp is `nil`.  It is when resp.StatusCode is not 200 OK where resp is non-nil and needs to be closed.

Not sure where you are copying this code from, we should fix that code @poornas